### PR TITLE
Fix local bind and support multiple readers and writers to a port

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: internal/binding/local.go [hunk 1]
```diff
@@ -412,127 +417,196 @@ func (lb *LocalBind) createATE(_ context.Context, ate *opb.Device, portMgr *ateP
...
+	dutID string
 }
 
-// dutPortMgr implements lemming's port creator interface.
-// Note: this interface is different from magna, to avoid making the two project depending on eachother.
-type dutPortMgr struct {
-	ports map[string]*chanPort
-	// TODO: lastCreatedPort is a hack because we don't currently model hardware ports correctly in lemming.
-	lastCreatedPort *chanPort
+// CreatePort impplements lemming's API for creating port.
```
Comment (guoshiuan) [inline internal/binding/local.go:528]: implements

File: internal/binding/local.go [hunk 2]
```diff
@@ -412,127 +417,196 @@ func (lb *LocalBind) createATE(_ context.Context, ate *opb.Device, portMgr *ateP
...
+			p.mu.RUnlock()
+		}
+	}()
 }
 
-// atePortMgr implement magna's port creator interface.
-// Note: this interface is different from lemming, to avoid making the two project depending on eachother.
-type atePortMgr struct {
-	ports map[string]*chanPort
+type portHandle struct {
```
Comment (guoshiuan) [inline internal/binding/local.go:580]: Can we have comment on this struct? look to me this is only for packet capture?

**Review by guoshiuan — APPROVED** [review]

File: internal/binding/local.go (hunk 1 above)
Comment (DanG100) [reply internal/binding/local.go]: done

File: internal/binding/local.go (hunk 2 above)
Comment (DanG100) [reply internal/binding/local.go:583]: done

**Pull request merged**